### DISCUSSION TITLE
How to mount longhorn RWX with nfs vers 4.2?

### DISCUSSION THREAD

**Discussion by pfaelzerchen** [longhorn/longhorn#7359] [Q&A]
Sorry, I am relatively new to this. By default, an RWX volume is mounted with nfs vers=4.1. This nfs version does not support setfattr on the mounted volume. As I want to expose some RWX volumes with samba and the samba-operator needs to set file attributes, it seems that I have to mount the shares at least with version 4.2.

I tried different this:
- creating a pv with mount option vers=4.2, then the pvc and then mounting it
- creating an additional storage class with mount option vers=4.2
- creating an additional storage class with nfsoptions (include copying the current default ones)

In any case, the pod cannot mount the volume. Going back to the longhorn default settings: Everything works fine.

I am using longhorn 1.5.3.

Does anyone have an idea?

**Comment by pfaelzerchen** [comment]
Solved this by migrating to rook-ceph.

**Comment by linuxfreakus** [comment]
I would like to know why the nfsOptions are not being used also.

I have longhorn 1.5.3 at the moment... if I detach the volume and re-attach so that the share manager restarts... it still says nfs 4.1 when I want to use 4.2
